### ISSUE TITLE
Align checkpoint selection with the supervised bridge objective

### ISSUE THREAD

**Issue by Ayberkrk** [Ayberkrk/cauren#6]
## Problem

When a dataset provides a real supervised target such as `deck_drop_5yr`, training adds a supervised BCE term to the loss and records validation classification accuracy. However, the best checkpoint is still selected only from validation reconstruction loss.

This can select an epoch with better reconstruction but worse deterioration prediction, even though the bridge model is explicitly being evaluated on a supervised forecasting task.

## Current behavior

In `cauren_core/training.py`, the total training loss can include:

```python
loss = recon_loss + 0.05 * aux_loss + 0.3 * supervised_loss
```

Validation also records supervised accuracy when labels are available.

The checkpoint selection logic, however, updates `best_state_dict` only when `val_avg_loss` improves. `val_avg_loss` is derived from reconstruction error, so the supervised metric does not participate in model selection.

This also means `best_validation_supervised_accuracy` currently refers to the supervised accuracy at the epoch selected by reconstruction loss, not necessarily the best supervised epoch.

## Proposed fix

Make checkpoint selection aware of the training objective.

For datasets without a real supervised target, keep the current reconstruction based selection behavior.

For datasets with a supervised target, use a supervised validation objective as the primary selection criterion, or use an explicitly documented combined criterion that reflects both supervised prediction and representation quality.

Accuracy alone may not be the best selection metric because the current bridge target is imbalanced. A validation metric such as BCE, balanced accuracy, F1, PR AUC, or another clearly justified metric would be preferable to majority sensitive accuracy alone.

## Acceptance criteria

- Unsupervised datasets continue to select checkpoints using reconstruction validation loss.
- Supervised datasets select checkpoints using a metric that includes the supervised objective.
- The chosen selection metric is stored in the training summary together with its best value and epoch.
- Reconstruction metrics are still reported separately for diagnostics.
- `best_validation_supervised_accuracy` or its replacement has an unambiguous meaning and is not simply the accuracy at an epoch chosen by an unrelated metric.
- Tests cover checkpoint selection for both supervised and unsupervised training modes.
- README evaluation text documents the checkpoint selection rule used for the bridge model.

A follow up evaluation step should still run the final selected checkpoint once on the untouched test split. This issue is specifically about selecting the checkpoint without using a metric that is disconnected from the supervised bridge objective.
